src/api.jl

```diff
@@ -899,31 +899,31 @@ end
 function h5t_get_array_dims(dtype_id, dims)
     var"#status#" = ccall((:H5Tget_array_dims2, libhdf5), Cint, (hid_t, Ptr{hsize_t}), dtype_id, dims)
     var"#status#" < 0 && error("Error getting dimensions of array")
-    return var"#status#"
+    return Int(var"#status#")
 end
 
 function h5t_get_array_ndims(dtype_id)
     var"#status#" = ccall((:H5Tget_array_ndims, libhdf5), Cint, (hid_t,), dtype_id)
     var"#status#" < 0 && error("Error getting ndims of array")
-    return var"#status#"
+    return Int(var"#status#")
 end
 
 function h5t_get_class(dtype_id)
     var"#status#" = ccall((:H5Tget_class, libhdf5), Cint, (hid_t,), dtype_id)
     var"#status#" < 0 && error("Error getting class")
-    return var"#status#"
+    return Int(var"#status#")
 end
 
 function h5t_get_cset(dtype_id)
     var"#status#" = ccall((:H5Tget_cset, libhdf5), Cint, (hid_t,), dtype_id)
     var"#status#" < 0 && error("Error getting character set encoding")
-    return var"#status#"
+    return Int(var"#status#")
 end
 
 function h5t_get_ebias(dtype_id)
     var"#status#" = ccall((:H5Tget_ebias, libhdf5), Csize_t, (hid_t,), dtype_id)
     var"#status#" < 0 && error("Error getting datatype floating point exponent bias")
```

Comment:
Same here
check should be <= 0 
https://portal.hdfgroup.org/display/HDF5/H5T_GET_EBIAS